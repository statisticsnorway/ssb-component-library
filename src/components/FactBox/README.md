FactBox
========

> Wrapper for accordion

### Usage

#### HTML

```html
<div class="ssb-fact-box ">
    <div class="ssb-accordion without-borders">
        <button class="accordion-header closed" onclick="{toggle classname to 'open'}" tabindex="0">
            <span class="button-grid">
          		<h5 class="ssb-title header-text no-margin">This is a standard accordion</h5>
          		<i class="expand-icon">{feather.chevronDown 20px}</i>
        	</span>
        </button>
        <div class="accordion-body closed">
			{insert content}
        </div>
    </div>
</div>
```

#### React

```jsx harmony
<FactBox header="This is a factbox header" text="This is paragraph text which explains the accordion" />
```

Available props:

| Name       | Type           | Description  |
| ---------- | ------------- | ----- |
| className   | string | Optional container class|
| header | string | Header text |
| text | string or element | Content in fact box |
| openByDefault | bool | Open when rendered. Defaults to false |

__Dependencies__
 - [Accordion](../Accordion)
