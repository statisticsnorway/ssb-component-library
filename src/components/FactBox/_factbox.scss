.ssb-fact-box {
  border: 1px solid $ssb-green-4;
  margin: 16px 0;
  padding: 20px 16px;
  width: 100%;
}
