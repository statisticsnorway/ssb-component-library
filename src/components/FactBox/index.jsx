import React from 'react';
import PropTypes from 'prop-types';
import Accordion from '../Accordion';

const FactBox = ({
	className, header, openByDefault, text,
}) => (
	<div className={`ssb-fact-box ${className}`}>
		<Accordion header={header} openByDefault={openByDefault} withoutBorders>
			{text}
		</Accordion>
	</div>
);

FactBox.defaultProps = {
	className: '',
	openByDefault: false,
};

FactBox.propTypes = {
	className: PropTypes.string,
	header: PropTypes.string.isRequired,
	openByDefault: PropTypes.bool,
	text: PropTypes.oneOfType([
		PropTypes.string,
		PropTypes.node,
	]).isRequired,
};

export default FactBox;
