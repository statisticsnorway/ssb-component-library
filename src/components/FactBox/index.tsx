import React, { useEffect, type ReactNode } from 'react'
import Accordion from '../Accordion'

interface FactBoxProps {
  className?: string
  header: string
  openByDefault?: boolean
  text: number | ReactNode
}

const FactBox: React.FC<FactBoxProps> = ({ className, header, openByDefault = false, text }) => {
  useEffect(() => {
    console.warn(
      'Warning: FactBox is deprecated and will be removed in a future release. Please use ExpansionBox instead.'
    )
  }, [])

  return (
    <div className={`ssb-fact-box${className ? ` ${className}` : ''}`}>
      <Accordion header={header} openByDefault={openByDefault} withoutBorders>
        {text}
      </Accordion>
    </div>
  )
}

export default FactBox
