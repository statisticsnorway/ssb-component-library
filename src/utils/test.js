import fs from 'fs';
import path from 'path';
import { render as rtlRender } from '@testing-library/react';

export * from '@testing-library/react';
export * from '@testing-library/jest-dom';

const styles = fs.readFileSync(path.relative(process.cwd(), './lib/bundle.css'), 'utf8');

export function render(ui, options) {
	const view = rtlRender(ui, options);

	const styleElement = document.createElement('style');
	styleElement.innerHTML = styles;
	document.body.appendChild(styleElement);

	return view;
}
