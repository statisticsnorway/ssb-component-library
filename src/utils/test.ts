import fs from 'fs'
import path from 'path'
import { RenderOptions, RenderResult, render as rtlRender } from '@testing-library/react'
import { ReactNode } from 'react'

export * from '@testing-library/react'

const styles = fs.readFileSync(path.relative(process.cwd(), './lib/bundle.css'), 'utf8')

export function render(ui: ReactNode, options?: RenderOptions): RenderResult {
  const view = rtlRender(ui, options)

  const styleElement = document.createElement('style')
  styleElement.innerHTML = styles
  document.body.appendChild(styleElement)

  return view
}
