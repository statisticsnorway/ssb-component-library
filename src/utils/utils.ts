// utils.ts
const classNames = (...classes: (string | boolean | undefined)[]): string => classes.filter(Boolean).join(' ')

export default classNames
