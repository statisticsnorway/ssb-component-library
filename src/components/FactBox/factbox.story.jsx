import React from 'react'
import FactBox from './index'

const mockedHeader = 'This is a fact box header'
const mockedText =
  'This is a normal paragraph text where the explanation of the fact is written in. ' +
  'Here you can write a concise and descriptive text that helps the user understand the fact. '

const mockedContent = <span>Insert custom HTML if you want to display something else than just text</span>

export default {
  title: 'FactBox',
}

export const Default = () => (
  <div style={{ width: '450px' }}>
    <FactBox header={mockedHeader} text={mockedText} />
    <FactBox header={mockedHeader} text={mockedContent} openByDefault />
  </div>
)
