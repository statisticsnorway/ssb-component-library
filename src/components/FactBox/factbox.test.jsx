import React from 'react'
import { screen, render } from '../../utils/test'
import FactBox from './index'

describe('FactBox component', () => {
  test('Matches the snapshot', () => {
    const { asFragment } = render(<FactBox header='fact box header' text='fact box Text' />)
    expect(asFragment()).toMatchSnapshot()
  })
  test('set openByDefault', async () => {
    const text = 'fact box Text'
    render(<FactBox header='fact box header' text={text} openByDefault />)

    expect(await screen.findByText(text)).toBeVisible()
  })
})
