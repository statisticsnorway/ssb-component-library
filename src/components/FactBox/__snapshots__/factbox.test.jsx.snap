// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FactBox component Matches the snapshot 1`] = `
<div
  className="ssb-fact-box "
>
  <Accordion
    header="fact box header"
    openByDefault={false}
    tabIndex={0}
    withoutBorders={true}
  >
    fact box Text
  </Accordion>
</div>
`;
